sarc/cli/db/prometheus/backup.py

```diff
@@ -0,0 +1,220 @@
+import json
+import logging
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Any
+
+from pydantic import BaseModel
+from tqdm import tqdm
+
+from sarc.client.job import _jobs_collection, JobStatistics, SlurmJob
+from sarc.config import config, TZLOCAL
+from sarc.jobs.sacct import update_allocated_gpu_type_from_nodes
+from sarc.jobs.series import (
+    _get_job_time_series_data_cache_key,
+    JOB_STATISTICS_METRIC_NAMES,
+)
+
+logger = logging.getLogger(__name__)
+
+
+DATE_FORMAT_HOUR = "%Y-%m-%dT%H:%M"
+
+
+class JobPrometheusData(BaseModel):
+    """Helper class to store Prometheus metrics for a job"""
+
+    cluster_name: str
+    job_id: int
+    submit_time: datetime
+    stored_statistics: JobStatistics | None = None
+    gpu_type: str | None = None
+
+
+@dataclass
+class DbPrometheusBackup:
+    """
+    Backup Prometheus metrics from database into cache.
+    Currently, save:
+    - SlurmJob.stored_statistics,
+      if not null and not empty.
+    - SlurmJob.allocated_gpu_type,
+      if GPU cannot be inferred from sacct or job nodes.
+    """
+
+    def execute(self) -> int:
+        cfg = config("scraping")
+        # Find clusters with prometheus_url
+        prometheus_cluster_names: list[str | None] = [
+            cluster.name for cluster in cfg.clusters.values() if cluster.prometheus_url
+        ]
+        if not prometheus_cluster_names:
+            logger.error(
+                "No cluster with prometheus_url configured, "
+                "we don't expect any job to have Prometheus metrics."
+            )
+            return -1
```

Comment:
There might have been some data in the past even if there is none right now, so we should still try to dump in that case.